src/pybind/mgr/mgr_module.py

```diff
@@ -1226,6 +1254,13 @@ def open_db(self) -> Optional[sqlite3.Connection]:
         uri = f"file:///{self.MGR_POOL_NAME}:{self.module_name}/main.db?vfs=ceph";
         self.log.debug(f"using uri {uri}")
         db = sqlite3.connect(uri, check_same_thread=False, uri=True)
+        # if libcephsqlite reconnects, update the addrv for blocklist
+        with db:
+            cur = db.execute('SELECT json_extract(ceph_status(), "$.addr");')
+            (addrv,) = cur.fetchone()
+            assert addrv is not None
+            self.log.debug(f"new libcephsqlite addrv = {addrv}")
+            self._ceph_register_client("libcephsqlite", addrv, True)
```

Comment:
Since the `replace` functionality of  `_ceph_register_client()` is not safe,  just deregister and register the new address?